include/extractor/traffic_signals.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef OSRM_EXTRACTOR_TRAFFIC_SIGNALS_HPP
+#define OSRM_EXTRACTOR_TRAFFIC_SIGNALS_HPP
+
+#include "util/typedefs.hpp"
+#include <unordered_set>
+
+#include <boost/unordered_set.hpp>
+
+namespace osrm
+{
+namespace extractor
+{
+
+struct TrafficSignals
+{
+    std::unordered_set<NodeID> bidirectional_nodes;
+    std::unordered_set<std::pair<NodeID, NodeID>, boost::hash<std::pair<NodeID, NodeID>>>
+        unidirectional_segments;
+
+    inline bool hasSignal(NodeID from, NodeID to) const
```

Comment:
Nit. Not sure we follow such code style everywhere but it seems it should be HasSignal